src/core/mac/mac.cpp

```diff
@@ -250,8 +232,54 @@ otError Mac::ConvertBeaconToActiveScanResult(Frame *aBeaconFrame, otActiveScanRe
     return error;
 }
 
-void Mac::StartEnergyScan(void)
+otError Mac::UpdateScanChannel(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    do
+    {
+        mScanChannel++;
+        VerifyOrExit(mScanChannel <= OT_RADIO_CHANNEL_MAX, error = OT_ERROR_NOT_FOUND);
+
+    }
+    while ((mScanChannels & (1U << mScanChannels)) == 0);
```

Comment:
`(1U << mScanChannels)` -> `(1U << mScanChannel)`

Reply:
Good eye! I am not sure how it got in. I did fix this after testing.

Reply:
Hmmm... checking the PR it looks like when doing "git rebase (squash)" somehow two commits were dropped.